scripts/common_cmdline.sh

```diff
@@ -74,17 +78,37 @@ function compileFull()
     if [[ \
         ("$exit_code" -ne "$expected_exit_code" || \
             ( $expect_output -eq 0 && -n "$errors" ) || \
-            ( $expect_output -ne 0 && $expected_exit_code -eq 0 && $expect_output -ne 2 && -z "$errors" ))
+            ( $expect_output -ne 0 && $expected_exit_code -eq 0 && \
+              $expect_output -ne 2 && -z "$errors" )
+        )
     ]]
     then
-        printError "Unexpected compilation result:"
-        printError "Expected failure: $expected_exit_code - Expected warning / error output: $expect_output"
-        printError "Was failure: $exit_code"
-        echo "$errors"
+        printError "Test failure:"
+        printError "Exit code: $exit_code"
+        printError "Expected:  $expected_exit_code"
+        printError "==== Output ===="
+        echo $errors
+        printError "== Output end =="
+        printError ""
+        case "$expect_output" in
+            0)
+                printError "No output was expected."
+                ;;
+            1)
+                printError "Expected warnings or errors."
+                ;;
+            2)
+                printError "Expected possible warnings."
```

Comment:
With `expect_output=2` we accept any output so maybe it would be better to just not print anything here at all?